Sources/StreamChat/Database/DatabaseContainer.swift

```diff
@@ -41,7 +41,9 @@ class DatabaseContainer: NSPersistentContainer {
         let context = newBackgroundContext()
         context.automaticallyMergesChangesFromParent = true
         context.mergePolicy = NSMergeByPropertyObjectTrumpMergePolicy
-        context.localCachingSettings = localCachingSettings
+        context.performAndWait { [localCachingSettings] in
+            context.localCachingSettings = localCachingSettings
+        }
```

Comment:
I'd say just `perform` is OK here, no need to block the initialization for longer than needed.